src/testcases/plei.jl

```diff
@@ -0,0 +1,103 @@
+# Astronomy example from Hairer et al 1992 p.245. As Hairer writes, "a
+# celestial mechanics problem( which we call 'the Pleiades'): seven stars in
+# the plan with coordinates x_i,y_i and masses m_i = i". Time span from
+#t =0 to t= 3. This is a non-stiff problem.
+#
+# Reference solution and a more in depth description can be found in
+# https://archimede.dm.uniba.it/~testset/report/plei.pdf
+
+
+export plei
+
+# create an instance of TestCaseExplicit by filling in the ...:
+plei = let
+    tcname = :plei ::Symbol # name should be same as variable name (except for upper/lower case)
+
+    T = Float64 # the datatype used, probably Float64
+    Tarr = [Matrix, SparseMatrixCSC][2] # the container datatype used
+                                        # for the Jacobian and mass
+                                        # matrix, probably Matrix
+                                        # (default) or
+                                        # SparseMatrixCSC.
+    const N = 7
+    const dof = 4N::Int # degrees of freedom
+    dae = 0 ::Int  # index of DAE, ==0 for ODE
+    # stiffness of system, one of the three constants
+    stiffness = [nonstiff, mildlystiff, stiff][1]
+
+    const r = zeros(Float64,7,7)
+    ## the problem function
+    function fn!(t::T,y::Vector{T},dydt::Vector{T})
+        # The ode function dydt = f(t,y) modifying dydt in-place.
+        # Note that all indices of dydt need to be written!
+
+        #parameters
+        #println(r)
+        for i = 1:N
+          for j =1:N
+            if i != j
+              r[i,j] = ((y[i] -y[j])^2 + (y[i+N] -y[j+N])^2)^(3/2)
```

Comment:
Sure, storing in a matrix only makes sense if we only calculate `r[i,j]` for i>j, then use the symmetry that `r[i,j] = r[j,i]`. Since we don't, an array is unnecessary.